File: ios/TranscriptionManager.swift
import Foundation

public class TranscriptionManager {

    private let bleClient: BleClient

    public init(bleClient: BleClient) {
        self.bleClient = bleClient
    }

    public func enableRealtimeTranscription(device: Device) async -> Result<Void, PlaudError> {
        do {
            // Command 0xA4: Enable real-time transcription
            let command = Data([0x5A, 0x04, 0xA4, 0x01])
            bleClient.sendCommand(command: command)
            return .success(())
        } catch {
            return .failure(PlaudError(code: 7001, message: "Enable real-time transcription failed: \(error.localizedDescription)"))
        }
    }

    public func disableRealtimeTranscription(device: Device) async -> Result<Void, PlaudError> {
        do {
            // Command 0xA4: Disable real-time transcription
            let command = Data([0x5A, 0x04, 0xA4, 0x00])
            bleClient.sendCommand(command: command)
            return .success(())
        } catch {
            return .failure(PlaudError(code: 7002, message: "Disable real-time transcription failed: \(error.localizedDescription)"))
        }
    }

    public func transcribeRecording(device: Device, recording: Recording) async -> Result<String, PlaudError> {
        do {
            // This will call the cloud service (Volcano Engine) to transcribe the recording
            // Implementation will depend on the cloud API details
            return .success("Transcribed text will be returned here")
        } catch {
            return .failure(PlaudError(code: 7003, message: "Transcribe recording failed: \(error.localizedDescription)"))
        }
    }

    public func summarizeRecording(device: Device, recording: Recording) async -> Result<String, PlaudError> {
        do {
            // This will call the cloud service (Volcano Engine) to summarize the recording
            // Implementation will depend on the cloud API details
            return .success("Summary will be returned here")
        } catch {
            return .failure(PlaudError(code: 7004, message: "Summarize recording failed: \(error.localizedDescription)"))
        }
    }
}

File: android/src/main/java/com/memotalk/sdk/TranscriptionManager.kt
package com.memotalk.sdk

import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

class TranscriptionManager(private val bleClient: BleClient) {

    suspend fun enableRealtimeTranscription(device: Device): Result<Unit, PlaudError> = withContext(Dispatchers.IO) {
        return@withContext try {
            // Command 0xA4: Enable real-time transcription
            val command = ByteArray(4)
            command[0] = 0x5A // Frame header
            command[1] = 0x04 // Length
            command[2] = 0xA4 // Command
            command[3] = 0x01 // Enable (1=enable, 0=disable)

            bleClient.sendCommand(command)
            Result.success(Unit)
        } catch (e: Exception) {
            Result.failure(PlaudError(7001, "Enable real-time transcription failed: ${e.message}"))
        }
    }

    suspend fun disableRealtimeTranscription(device: Device): Result<Unit, PlaudError> = withContext(Dispatchers.IO) {
        return@withContext try {
            // Command 0xA4: Disable real-time transcription
            val command = ByteArray(4)
            command[0] = 0x5A // Frame header
            command[1] = 0x04 // Length
            command[2] = 0xA4 // Command
            command[3] = 0x00 // Disable

            bleClient.sendCommand(command)
            Result.success(Unit)
        } catch (e: Exception) {
            Result.failure(PlaudError(7002, "Disable real-time transcription failed: ${e.message}"))
        }
    }

    suspend fun transcribeRecording(device: Device, recording: Recording): Result<String, PlaudError> = withContext(Dispatchers.IO) {
        return@withContext try {
            // This will call the cloud service (Volcano Engine) to transcribe the recording
            // Implementation will depend on the cloud API details
            Result.success("Transcribed text will be returned here")
        } catch (e: Exception) {
            Result.failure(PlaudError(7003, "Transcribe recording failed: ${e.message}"))
        }
    }

    suspend fun summarizeRecording(device: Device, recording: Recording): Result<String, PlaudError> = withContext(Dispatchers.IO) {
        return@withContext try {
            // This will call the cloud service (Volcano Engine) to summarize the recording
            // Implementation will depend on the cloud API details
            Result.success("Summary will be returned here")
        } catch (e: Exception) {
            Result.failure(PlaudError(7004, "Summarize recording failed: ${e.message}"))
        }
    }
}
